components/perf/dynamicRendering/src/lazyContent.js

```diff
@@ -1,73 +1,22 @@
-import React, {Component} from 'react'
+import React from 'react'
 import PropTypes from 'prop-types'
-
-export class LazyContent extends Component {
-  constructor(props) {
-    super(props)
-    this.refTarget = React.createRef()
-  }
-
-  _observer = null
-
-  state = {
-    isIntersecting: false
-  }
-
-  handleChange = ([{isIntersecting}]) => {
-    if (isIntersecting) {
-      this.unobserve()
-      this._observer = null
-      this.setState({isIntersecting})
-    }
-  }
-
-  unobserve = () => {
-    try {
-      this._observer.unobserve(this.refTarget.current)
-    } catch (e) {
-      console.warn(e)
-    }
-  }
-
-  _startIntersectionObserver = () => {
-    this._observer = new window.IntersectionObserver(this.handleChange)
-    try {
-      this._observer.observe(this.refTarget.current, {
-        rootMargin: this.props.rootMargin
-      })
-    } catch (e) {
-      console.warn(e)
-    }
+import {useNearScreen} from '@schibstedspain/sui-react-hooks'
+
+export default function LazyContent({
+  children,
+  rootMargin = '100px 0px 0px',
+  placeholder,
+  height
+}) {
+  const [isNearScreen, fromRef] = useNearScreen({offset: rootMargin})
+
+  if (isNearScreen) {
```

Comment:
A bit late, but instead of :
```
if (isNearScreen) {
    return children
  } else if (placeholder) {
    return <div ref={fromRef}>{placeholder}</div>
  } else {
    return <div ref={fromRef} style={{height: `${height}px`}} />
  }
```

What do you think about:
```
if (isNearScreen) {return children}

if (placeholder) {
    return <div ref={fromRef}>{placeholder}</div>
}

return <div ref={fromRef} style={{height: `${height}px`}} />
```

Since we are executing a `return`, the `else statement` only adds mental complexity when evaluating the cases. @midudev

Reply:
I avoided to touch the return statement from the render in order to focus only on the things I wanted to change. :)